packages/react-components/react-rating-preview/src/components/RatingItem/RatingItem.test.tsx

```diff
@@ -1,10 +1,108 @@
 import * as React from 'react';
-import { render } from '@testing-library/react';
+import { queryAllByRole, render } from '@testing-library/react';
+import { isConformant } from '../../testing/isConformant';
 import { RatingItem } from './RatingItem';
+import { RatingItemProvider } from '../../contexts/RatingItemContext';
+import { RatingItemContextValue } from './RatingItem.types';
+import { StarFilled, StarRegular } from '@fluentui/react-icons';
 
 describe('RatingItem', () => {
-  it('renders a default state', () => {
-    const result = render(<RatingItem>Default RatingItem</RatingItem>);
-    expect(result.container).toMatchSnapshot();
+  isConformant({
+    Component: RatingItem,
+    displayName: 'RatingItem',
+    // Need to disable this test because certain slots are not rendered without specific context values
+    disabledTests: ['component-has-static-classnames-object'],
+  });
+  it('does not render input elements when interactive is false', () => {
+    const contextValues: RatingItemContextValue = {
+      value: 3,
+      interactive: false,
+      iconFilled: <StarFilled />,
+      iconOutline: <StarRegular />,
+      color: 'neutral',
+      step: 1,
+      size: 'medium',
+    };
+    const { getByTestId } = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={2} data-testid="test" />
+      </RatingItemProvider>,
+    );
+    const items = getByTestId('test');
+    expect(items.getElementsByTagName('input').length).toEqual(0);
+  });
+  it('renders only the full value input when step is 1 and interactive is true', () => {
+    const contextValues: RatingItemContextValue = {
+      value: 3,
+      interactive: true,
+      iconFilled: <StarFilled />,
+      iconOutline: <StarRegular />,
+      color: 'neutral',
+      step: 1,
+      size: 'medium',
+    };
+    const { getByTestId } = render(
+      <RatingItemProvider value={contextValues}>
+        <RatingItem value={2} data-testid="test" />
+      </RatingItemProvider>,
+    );
+    const items = getByTestId('test');
+    expect(items.getElementsByTagName('input').length).toEqual(1);
```

Comment:
Use `getByRole`. And maybe verify the input's value (not sure if you need to use `getAttribute('value')`)

```suggestion
    const { getByRole } = render(
      <RatingItemProvider value={contextValues}>
        <RatingItem value={2} />
      </RatingItemProvider>,
    );
    const input = getByRole('radio') as HTMLInputElement;
    expect(input.value).toEqual(2);
```